simulation/entities_test.py

```diff
@@ -19,77 +27,75 @@ def test_has_enough_conviction(self):
         Test that the function works. If we set the probability to 1, Participant should buy in.
         If we set the probability to 0, 0 will be returned.
         """
-        with patch('entities.probability') as mock:
-            mock.return_value = True
-            # Must set Participant's sentiment artificially high, because of Zargham's force calculation
-            self.p.sentiment = 1
-            delta_holdings = self.p.buy()
-            self.assertGreater(delta_holdings, 0)
-
-        with patch('entities.probability') as mock:
-            mock.return_value = False
-            delta_holdings = self.p.buy()
-            self.assertEqual(delta_holdings, 0)
+
+        # Must set Participant's sentiment artificially high, because of Zargham's force calculation
+        self.p.sentiment = 1
+        self.p._probability = always
+        delta_holdings = self.p.buy()
+        self.assertGreater(delta_holdings, 0)
+
+        self.p._probability = never
+        delta_holdings = self.p.buy()
+        self.assertEqual(delta_holdings, 0)
 
     def test_sell(self):
         """
         Test that the function works. If we set the probability to 1, Participant should buy in.
         If we set the probability to 0, 0 will be returned.
         """
 
-        with patch('entities.probability') as mock:
-            mock.return_value = True
-            self.p.sentiment = 0.1
-            delta_holdings = self.p.sell()
-            self.assertLess(delta_holdings, 0)
+        self.p._probability = always
+        self.p.sentiment = 0.1
+        delta_holdings = self.p.sell()
+        self.assertLess(delta_holdings, 0)
 
-        with patch('entities.probability') as mock:
-            mock.return_value = False
-            delta_holdings = self.p.sell()
-            self.assertEqual(delta_holdings, 0)
+        self.p._probability = never
+        delta_holdings = self.p.sell()
+        self.assertEqual(delta_holdings, 0)
 
     def test_create_proposal(self):
         """
         Test that the function works. If we set the probability to 1, we should
         get True. If not, we should get False.
         """
-        with patch('entities.probability') as mock:
-            mock.return_value = True
-            self.assertTrue(self.p.create_proposal(10000, 0.5, 100000))
 
-        with patch('entities.probability') as mock:
-            mock.return_value = False
-            self.assertFalse(self.p.create_proposal(10000, 0.5, 100000))
+        self.p._probability = always
+        self.assertTrue(self.p.create_proposal(10000, 0.5, 100000))
+
+        self.p._probability = never
+        self.assertFalse(self.p.create_proposal(10000, 0.5, 100000))
 
     def test_vote_on_candidate_proposals(self):
         """
         Test that the function works. If we set the probability to 1, we should
         get a dict of Proposal UUIDs that the Participant would vote on. If not,
         we should get an empty dict.
         """
-
+        
```

Comment:
i suggest pylint. VSCode, whatever IntelliJ sells (what was it again, it was good)